fmralign/alignment_methods.py

```diff
@@ -0,0 +1,252 @@
+    else:
+        sc = 1
+    return R.T, sc
+
+
+def optimal_permutation(X, Y):
+    """Compute the optmal permutation matrix of X toward Y
+    Parameters
+    ----------
+    X: (n_samples, n_features) nd array
+        source data
+    Y: (n_samples, n_features) nd array
+        target data
+
+    Returns
+    ----------
+    permutation : (n_features, n_features) nd array
+        transformation matrix
+    """
+    dist = pairwise_distances(X.T, Y.T)
+    u = linear_assignment(dist)
+    permutation = scipy.sparse.csr_matrix(
+        (np.ones(X.shape[1]), (u[:, 0], u[:, 1]))).T
+    return permutation
+
+
+class Alignment(BaseEstimator, TransformerMixin):
+    def __init__(self):
+        pass
+
+    def fit(self, X, Y):
+        pass
+
+    def transform(self, X):
+        pass
+
+
+class Identity(Alignment):
+    """The simplest kind of alignment to be used as a baseline for benchmarks. RX = X
+    """
+
+    def transform(self, X):
+        """returns X"""
+        return X
+
+
+class ScaledOrthogonalAlignment(Alignment):
+    """Compute a mixing matrix R and a scaling sc such that
+    frobenius norm ||sc RX - Y||^2 is minimized and
+    R is an orthogonal matrix
+
+    Parameters
+    ---------
+    scaling : boolean, optional
+        Determines whether a scaling parameter is applied to improve transform.
+    R : ndarray (n_features, n_features)
+        Optimal orthogonal transform
+    """
+
+    def __init__(self, scaling=True):
+        self.scaling = scaling
+        self.scale = None
+
+    def fit(self, X, Y):
+        """ Fit orthogonal R s.t. ||sc XR - Y||^2
+        ----------
+        X: (n_timeframes, n_features) nd array
+            source data
+        Y: (n_timeframes, n_features) nd array
+            target data
+        """
+        R, sc = scaled_procrustes(X, Y, scaling=self.scaling)
+        self.scale = sc
+        self.R = sc * R
+        return self
+
+    def transform(self, X):
+        """Transform X using optimal transform computed during fit.
+        """
+        return self.scale * X.dot(self.R)
+
+
+class RidgeAlignment(Alignment):
+    """ Compute an scikit-estimator R using a mixing matrix M such that
+    frobenius norm || XM - Y ||^2 + alpha ||M||^2 is minimized with built-in cross-validation
+
+    Parameters
+    ----------
+    R : scikit-estimator from sklearn.linear_model.RidgeCV with method fit, predict
+    alpha : numpy array of shape [n_alphas]
+        Array of alpha values to try. Regularization strength; must be a positive float. Regularization
+        improves the conditioning of the problem and reduces the variance of
+        the estimates. Larger values specify stronger regularization.
+        Alpha corresponds to ``C^-1`` in other linear models.
+    cv : int, cross-validation generator or an iterable, optional
+        Determines the cross-validation splitting strategy. Possible inputs for cv are:
+        -None, to use the efficient Leave-One-Out cross-validation
```

Comment:
- None,

Reply:
I don't get what you mean here @bthirion